### PR TITLE
DOC: Generate the list of supported fonts dynamically from codes

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Similar to #3409, but for the list of supported fonts. Changes in `doc/conf.py` and `doc/_static/style.css` are the same as in #3409.

**Preview**: 

- This branch: https://pygmt-dev--3406.org.readthedocs.build/en/3406/techref/fonts.html
- Main branch: https://www.pygmt.org/dev/techref/fonts.html

### CODE REVIEW

**Review by michaelgrund — APPROVED** [review]

File: doc/techref/fonts.md [hunk 1]
```diff
@@ -7,25 +11,105 @@ either `"Helvetica"` or `"0"`. For the special fonts "Symbol" (**12**) and
...
+
+```{code-cell}
+---
+tags: [remove-input]
+---
+"""
+Script to generate visual samples of the fonts.
+"""
+import pygmt
+
```
Comment (yvonnefroehlich) [inline doc/techref/fonts.md:76]:
```suggestion
```

File: doc/techref/fonts.md [hunk 2]
```diff
@@ -7,25 +11,105 @@ either `"Helvetica"` or `"0"`. For the special fonts "Symbol" (**12**) and
...
+for i, font in enumerate(fonts):
+    x0 = x1 if i < 17 else x2
+    y0 = i % 17
+    font_no, font_name = i, font
+
+    # Deal with special cases
+    if font in ["Symbol", "ZapfDingbats"]:
+        font_name = f"{font} @%0%({font})@%%"
+    if font == "ZapfDingbats":
+        font_no = f"@%0%34@%%"
```
Comment (yvonnefroehlich) [inline doc/techref/fonts.md:104]:
```suggestion
        font_no = "@%0%34@%%"
```

File: doc/techref/fonts.md [hunk 3]
```diff
@@ -7,25 +11,105 @@ either `"Helvetica"` or `"0"`. For the special fonts "Symbol" (**12**) and
...
+---
+tags: [remove-input]
+---
+"""
+Script to generate visual samples of the fonts.
+"""
+import pygmt
+
+
+x1, x2, dx = 0.0, 7.0, 0.75
```
Comment (yvonnefroehlich) [inline doc/techref/fonts.md:78]:
```suggestion
x1, x2, dx = 0, 7, 0.75
```

File: doc/techref/fonts.md [hunk 4]
```diff
@@ -7,25 +11,105 @@ either `"Helvetica"` or `"0"`. For the special fonts "Symbol" (**12**) and
...
+fig = pygmt.Figure()
+# Draw the table
+fig.basemap(region=[-0.5, 14, -1.5, 18], projection="X14c/-10c", frame=0)
+fig.plot(x=[-0.5, 14], y=[-0.5, -0.5])
+for x in (0.5, 6.5, 7.5):
+    fig.plot(x=[x, x], y=[-1.5, 18])
+# Table header
+fig.text(
+    x=[x1, x1 + dx, x2, x2 + dx],
+    y=[-1.0] * 4,
```
Comment (yvonnefroehlich) [inline doc/techref/fonts.md:89]:
```suggestion
    y=[-1] * 4,
```

File: doc/techref/fonts.md [hunk 5]
```diff
@@ -7,25 +11,105 @@ either `"Helvetica"` or `"0"`. For the special fonts "Symbol" (**12**) and
...
+    text=["#", "Font Name"] * 2,
+    justify=["MC", "ML"] * 2,
+    font="Helvetica-Bold",
+)
+# Fonts
+for i, font in enumerate(fonts):
```
Comment (yvonnefroehlich) [inline doc/techref/fonts.md:95]: Good to know that, it is possible to use variables defined in one code-cell in the following code-cells.

File: doc/techref/fonts.md [hunk 6]
```diff
@@ -7,25 +11,105 @@ either `"Helvetica"` or `"0"`. For the special fonts "Symbol" (**12**) and
...
+    "NewCenturySchlbk-BoldItalic",
+    "Palatino-Roman",
+    "Palatino-Italic",
+    "Palatino-Bold",
+    "Palatino-BoldItalic",
+    "ZapfChancery-MediumItalic",
+    "ZapfDingbats",
+]
+
+text = "|Font No. | Font Name | Font No. | Font Name |\n"
```
Comment (yvonnefroehlich) [inline doc/techref/fonts.md:58]:
```suggestion
text = "| Font No. | Font Name | Font No. | Font Name |\n"
```

File: doc/techref/fonts.md [hunk 7]
```diff
@@ -7,25 +11,105 @@ either `"Helvetica"` or `"0"`. For the special fonts "Symbol" (**12**) and
...
+text += "|:---:|:---|:---:|:---|\n"
+for i in range(17):
+    j = i + 17
+    text += f"| {i} | {fonts[i]} | {j} | {fonts[j]} |\n"
+text += f"| | | 34 | {fonts[34]}\n"
```
Comment (yvonnefroehlich) [inline doc/techref/fonts.md:63]:
```suggestion
text += f"| | | 34 | {fonts[34]} |\n"
```

**Review by yvonnefroehlich — APPROVED** [review]
Cool idea to generate the table with the font samples directly in PyGMT 🙂!

**Comment by seisman** [comment]
Need to merge #3418 first.

**Pull request merged**